### PR TITLE
fix index bug in exchange removeOrder

### PR DESCRIPTION
**Pull request by treer-euedge**
Fixes bug: index in active order lists is not updated in Exchange removeOrder methods

### CODE REVIEW

File: contracts/Exchange.sol [hunk 1]
```diff
@@ -235,18 +235,18 @@ contract Exchange is Restricted {
     }
 
     function _removeBuyOrder(Order storage order) private {
-        _removeOrder(activeBuyOrders, order.index);
+        if (order.index < activeBuyOrders.length - 1) {
+            activeBuyOrders[order.index] = activeBuyOrders[activeBuyOrders.length - 1];
+            buyTokenOrders[activeBuyOrders[order.index]].index = order.index;
```
Comment (phraktle) [inline contracts/Exchange.sol]: why not just update `order.index`? `order` is already a storage pointer...

Comment (phraktle) [inline contracts/Exchange.sol]: might be better to memorize activeBuyOrders[order.index]

Comment (phraktle) [reply contracts/Exchange.sol]: similarly, `uint lastIndex = activeBuyOrders.length - 1'

Comment (treer-euedge) [reply contracts/Exchange.sol]: done

**Pull request merged**